packages/react/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,331 @@
+        return
+      }
+
+      const { itemsForRender } = this.state
+      const elementToBeFocused = itemsForRender[parent['id']].elementRef
+
+      if (!elementToBeFocused) {
+        return
+      }
+
+      elementToBeFocused.current.focus()
+      _.invoke(predefinedProps, 'onFocusParent', e, treeItemProps)
+    },
+    onFocusFirstChild: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+      const { id } = treeItemProps
+
+      const { itemsForRender } = this.state
+      const currentElement = itemsForRender[id].elementRef
+
+      if (!currentElement || !currentElement.current) {
+        return
+      }
+
+      const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current)
+
+      if (!elementToBeFocused) {
+        return
+      }
+
+      elementToBeFocused.focus()
+      _.invoke(predefinedProps, 'onFocusFirstChild', e, treeItemProps)
+    },
+    onSiblingsExpand: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+      if (this.props.exclusive) {
+        return
+      }
+
+      const { id, siblings } = treeItemProps
+      const { activeItemIds } = this.state
+
+      siblings.forEach(sibling => {
+        if (Tree.hasSubtree(sibling) && activeItemIds.indexOf(sibling['id']) < 0) {
+          activeItemIds.push(sibling['id'])
+        }
+      })
+
+      if (Tree.hasSubtree(treeItemProps) && activeItemIds.indexOf(id) < 0) {
+        activeItemIds.push(id)
+      }
+
+      this.setState({
+        activeItemIds,
+      })
+
+      _.invoke(predefinedProps, 'onSiblingsExpand', e, treeItemProps)
+    },
+  })
+
+  renderContent(): React.ReactElement[] {
+    const { activeItemIds, itemsForRender } = this.state
+    const { items, renderItemTitle } = this.props
+
+    if (!items) return null
+
+    const renderItems = (items: ShorthandCollection<TreeItemProps>): React.ReactElement[] => {
+      return items.reduce((renderedItems: any[], item: ShorthandValue<TreeItemProps>) => {
+        const itemForRender = itemsForRender[item['id']]
+        const { elementRef, ...restItemForRender } = itemForRender
+        const isSubtree = Tree.hasSubtree(item)
+        const isSubtreeOpen = isSubtree && activeItemIds.indexOf(item['id']) > -1
+
+        const renderedItem = TreeItem.create(item, {
+          defaultProps: {
+            className: Tree.slotClassNames.item,
+            open: isSubtreeOpen,
+            renderItemTitle,
+            key: item['id'],
+            ...restItemForRender,
+          },
+          overrideProps: this.handleTreeItemOverrides,
+        })
+
+        // Only need refs of the items that spawn subtrees, when they need to be focused
+        // by any of their children, using Arrow Left.
+        const finalRenderedItem = isSubtree ? (
+          <Ref key={item['id']} innerRef={elementRef}>
+            {renderedItem}
+          </Ref>
+        ) : (
+          renderedItem
+        )
+
+        return [
+          ...(renderedItems as any[]),
```

Comment:
```suggestion
          ...renderedItems,
```

Is not required after https://github.com/stardust-ui/react/pull/1779/files#r319034709